### ISSUE TITLE
Current instructions for authorising with GitHub don't work on MacOS

### ISSUE THREAD

**Issue by alexdewar** [ImperialCollegeLondon/grad_school_git_course#32]
I'm not sure if this issue affects all MacOS users, but certainly there were a few in the class today for whom the instructions on how to use the PAT seemed not to work. The instructions seemed to work fine on Windows, but I'm not sure if that's due to Git for Windows including some kind of a GitHub authentication tool.

My understanding of the whole PAT thing is somewhat poor, but I had it in mind that they should just be used in the same way as a password (i.e. when git asks for a username and password, you enter your GitHub username and paste the PAT)... right?

A few of them were happy enough using SSH instead and I was able to switch them over, but it was generally a fairly frustrating experience all round.

I suppose our options are:

1. Figure out how to make the PATs work on MacOS (and Linux...?)
2. Get MacOS/Linux users to use SSH and continue using PATs with the Windows users
3. Just get everyone to use SSH (Git for Windows does bundle ssh so it should work fine)

What does everyone think?

**Comment by dandavies99** [comment]
@alexdewar do you mean [these instructions](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-token)? I am pretty sure that should work for Mac. The PATs work as you describe, just use them as a password, so should be pretty system agnostic. 

Was the PAT just not working as a password? 

It's likely that the students in question weren't selecting "repo" when generating the token, or did you check this? I found that by the time they get to step 13 (!?) of those instructions, they've forgotten that's what they're meant to do. Maybe we should make it more obvious.

**Comment by alexdewar** [comment]
Yeah, I think so. I skipped over that part of the tutorial when I was looking through by myself, but on closer inspection it does seem like it should work.

I didn't check whether they were doing that, but that would make sense! I think my lack of understanding of PATs probably hampered things.

Maybe the fact that they were all MacOS users was a red herring. It doesn't seem like there's any sane reason why entering a password wouldn't work just the same as it does on Windows.
